dashboard-api/internal/platform/server/middleware/auth/auth_test.go

```diff
@@ -0,0 +1,144 @@
+package auth
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-playground/assert/v2"
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+func TestEnsureValidTokenMiddleware(t *testing.T) {
+	audience := "aud"
+	jwk := GenerateJWK(t)
+	testServer := SetupTestJWKServer(t, jwk)
+	defer testServer.Close()
+
+	t.Run("invalid token", func(t *testing.T) {
```

Comment:
I would make the name a bit more descriptive so we know _why_ the token is invalid (i.e. the audience is wrong)

Reply:
:100: will update it